[ONNX] Enable support for roll() op. (#58389)

BowenBao opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#58697 [ONNX] Enable support for roll() op. (#58389)**
* #58696 [ONNX] handle aten::_set_item on Dict in convertInplaceOpsAndTrackAlias (#58317)
* #58695 [ONNX] use consistent quoting for string literals (#57757)
* #58694 [ONNX] Improve lower tuples and handle control flow  (#57650)
* #58693 [ONNX] Update special post process for SequenceInsert after SequenceEmpty (#56965)
* #58692 Support symbolic for conv_tbc (#58359)
* #58691 [ONNX] RNN scripting (#57564)
* #58690 [ONNX] Update instance_norm2 symbolic to handle track_running_stats=True (#55051)

1. Add a symbolic function for aten::roll() op in symbolic_opset9.py.
2. Add a test with multiple scenarios as well.

Co-authored-by: fatcat-z <jiz@microsoft.com>

Differential Revision: [D28714807](https://our.internmc.facebook.com/intern/diff/D28714807)

fatcat-z committed:
[ONNX] Enable support for roll() op. (#58389)

1. Add a symbolic function for aten::roll() op in symbolic_opset9.py.
2. Add a test with multiple scenarios as well.

Co-authored-by: fatcat-z <jiz@microsoft.com>

[ghstack-poisoned]

BowenBao committed:
Update on "[ONNX] Enable support for roll() op. (#58389)"


1. Add a symbolic function for aten::roll() op in symbolic_opset9.py.
2. Add a test with multiple scenarios as well.

Co-authored-by: fatcat-z <jiz@microsoft.com>

[ghstack-poisoned]

SplitInfinity commented:
@splitinfinity has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28622443).

SplitInfinity commented:
@SplitInfinity has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28714807).

BowenBao committed:
Update on "[ONNX] Enable support for roll() op. (#58389)"


1. Add a symbolic function for aten::roll() op in symbolic_opset9.py.
2. Add a test with multiple scenarios as well.

Co-authored-by: fatcat-z <jiz@microsoft.com>

Differential Revision: [D28714807](https://our.internmc.facebook.com/intern/diff/D28714807)

[ghstack-poisoned]

SplitInfinity commented:
@SplitInfinity has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28714807).
